solutionbox/code_free_ml/mltoolbox/code_free_ml/data/local_predict.py

```diff
@@ -0,0 +1,251 @@
+      description='Runs local prediction')
+  # I/O args
+  parser.add_argument('--predict-data',
+                      required=True,
+                      help='Data to run prediction on')
+  parser.add_argument('--trained-model-dir',
+                      required=True,
+                      help='Usually train_output_path/model.')
+  parser.add_argument('--output-location',
+                      required=True,
+                      help=('Location to save output.'))
+  parser.add_argument('--output-format',
+                      default='csv',
+                      choices=['csv', 'json'],
+                      help=('format of prediction results.'))
+
+  # Other args
+  parser.add_argument('--batch-size',
+                      required=False,
+                      default=1000,
+                      type=int,
+                      help=('Batch size. Larger values consumes more memrory '
+                            'but takes less time to finish.'))
+  parser.add_argument('--shard-files',
+                      dest='shard_files',
+                      action='store_true',
+                      help='One prediction output file is made per input file.')
+  parser.add_argument('--no-shard-files',
+                      dest='shard_files',
+                      action='store_false',
+                      help='Don\'t shard files')
+  parser.set_defaults(shard_files=True)
+
+  args, _ = parser.parse_known_args(args=argv[1:])
+
+  return args
+
+
+def batch_csv_reader(csv_file, batch_size):
+  """Reads a csv file in batches.
+
+  Args:
+    csv_file: file path
+    batch_size: positive integer
+
+  Returns:
+    generator
+  """
+  counter = 0
+  batched_lines = []
+  for one_csv_line in file_io.FileIO(csv_file, 'r'):
+    batched_lines.append(one_csv_line.rstrip())
+    counter += 1
+    if counter == batch_size:
+      yield batched_lines
+      counter = 0
+      batched_lines = []
+
+  if batched_lines:
+    yield batched_lines
+
+
+def run_batch_prediction(session, input_alias_map, output_alias_map, batch_csv_strings):
+  """Runs a tensorflow prediction graph from batched csv input.
+
+  Args:
+    session: tensorflow session
+    input_alias_map: dict of {friendly_name: tensor_name} for feeding the
+        feed_dict. Should only have 1 value as the model takes csv data.
+    output_alias_map: dict of {friendly_name: tensor_name} for fetching the
+        ouput tensors form session.run
+    batch_csv_strings: list of csv strings to feed to the model.
+
+  Returns:
+    List of dicts in the form {output_alias_friendly_name: predicted_value}.
+  """
+  _, csv_tensor_name = input_alias_map.items()[0]
+  raw_result = session.run(fetches=output_alias_map,
+                           feed_dict={csv_tensor_name: batch_csv_strings})
+
+  # Convert from numpy arrays to python lists
+  result = {key: raw_result[key].flatten().tolist() for key in raw_result}
+
+  batch_size = len(batch_csv_strings)
+  json_list = []
```

Comment:
A little bit odd to me to call it "json_list" and "json_result" below. Maybe just "results"?